src/types.jl

```diff
@@ -151,41 +175,37 @@ function pgdata(::Type{PostgresType{:numeric}}, ptr::Ptr{UInt8}, data::Number)
     ptr = storestring!(ptr, string(data))
 end
 
-function pgdata(::PGStringTypes, ptr::Ptr{UInt8}, data::ByteString)
```

Comment:
These functions were here so that `ByteString`s could be stored directly and strings which don't have a `unsafe_convert(Ptr{UInt8}, str)` method would be converted to `ByteString`s first. I think the correct way to fix these functions is to have this first one accept `String` and have the other ones call `String(data)` first.